qiskit_nature/problems/second_quantization/lattice/lattice/triangular_lattice.py

```diff
@@ -0,0 +1,295 @@
+        )
+    return list_of_edges
+
+
+def _default_position(
+    size: Tuple[int, int], boundary_condition: BoundaryCondition
+) -> Dict[int, List[float]]:
+    """return a dictionary of default positions for visualization of a two-dimensional lattice.
+
+    Args:
+        size : Lengths of each dimension.
+        boundary_condition : Boundary condition for the lattice.
+                The available boundary conditions are:
+                BoundaryCondition.OPEN, BoundaryCondition.PERIODIC.
+    Returns:
+        Dict[int, List[float]] : The keys are the labels of lattice points,
+            and the values are two-dimensional coordinates.
+    """
+    pos = {}
+    width = 0.0
+    if boundary_condition == BoundaryCondition.PERIODIC:
+        # the positions are shifted along the x- and y-direction
+        # when the boundary condition is periodic.
+        # The width of the shift is fixed to 0.2.
+        width = 0.2
+    for index in range(np.prod(size)):
+        # maps an index to two-dimensional coordinate
+        # the positions are shifted so that the edges between boundaries can be seen
+        # for the periodic cases.
+        coord = np.array(divmod(index, size[0]))[::-1] + width * np.sin(
+            pi * np.array(divmod(index, size[0])) / (np.array(size)[::-1] - 1)
+        )
+        pos[index] = coord.tolist()
+    return pos
+
+
+class TriangularLattice(Lattice):
+    """Triangular lattice."""
+
+    def __init__(
+        self,
+        rows: int,
+        cols: int,
+        edge_parameter: Union[complex, Tuple[complex, complex, complex]] = 1.0,
+        onsite_parameter: complex = 0.0,
+        boundary_condition: BoundaryCondition = BoundaryCondition.OPEN,
+    ) -> None:
+        """
+        Args:
+            rows: Length of the x direction.
+            cols: Length of the y direction.
+            edge_parameter: Weights on the edges in x, y and diagonal directions.
+                This is specified as a tuple of length 3 or a single value.
+                When it is a single value, it is interpreted as a tuple of length 3
+                consisting of the same values.
+                Defaults to 1.0,
+            onsite_parameter: Weight on the self-loops, which are edges connecting a node to itself.
+                Defaults to 0.0.
+            boundary_condition: Boundary condition for the lattice.
+                The available boundary conditions are:
+                BoundaryCondition.OPEN, BoundaryCondition.PERIODIC.
+                Defaults to BoundaryCondition.OPEN.
+
+        Raises:
+            ValueError: Given size, edge parameter or boundary condition are invalid values.
+        """
+        self.rows = rows
+        self.cols = cols
+        self.size = (rows, cols)
+        self.dim = 2
+        self.boundary_condition = boundary_condition
+
+        if rows < 2 or cols < 2 or (rows, cols) == (2, 2):
```

Comment:
Isn't this the same as the following?
```suggestion
        if rows <= 2 and cols <= 2:
```
If so, the error message below could be improved. It should also mention _why_ this causes an error.

Reply:
On another note: do you check that these numbers are positive anywhere?

Reply:
In the original code, this is the case when either `rows` or `cols` is an integer less than 2, as well as when both `rows` and `cols` are 2. This is because a triangular lattice cannot be well defined with this size.
Examples include (rows, cols) = (1, 5) and (rows, cols) = (3, 0).
On the other hand, the proposed code does not include the above examples, which is not what I expected.
As for the second comment, I actually don't check `rows` and `cols` are positive or not, but I expect this condition automatically exclude 0 and negative integers.

Reply:
There may be a clearer way to write the code, but I haven't come up with it.